lib/chef/provider/package/windows.rb

```diff
@@ -34,9 +34,16 @@ class Windows < Chef::Provider::Package
 
         require 'chef/provider/package/windows/registry_uninstall_entry.rb'
 
+        def define_resource_requirements
+          requirements.assert(:install) do |a|
+            a.assertion { new_resource.source unless package_provider == :msi }
+            a.failure_message Chef::Exceptions::NoWindowsPackageSource, "Source for package #{new_resource.name} must be specified in the resource's source property for package to be installed because the package_name property is used to test for package installation state for this package type."
```

Comment:
Not seeing the missing "the" but to @btm's point, this is tough to get just right and make it concise. We could just say that the source is required and leave it at that but if the user uses a url/path as the package_name, then they might be confused why they need to specify a `source` which they don't have to do for MSIs.